### DISCUSSION TITLE
When does a webc:setup script run and can it access the data cascade?

### DISCUSSION THREAD

**Discussion by darthmall** [11ty/eleventy#2889] [Q&A]
I’ve only just begun to explore `webc:setup` and I’m a little uncertain as to when it is executed and to what data it has access. As an example, I have tried setting the `<title>` using data from a template’s frontmatter and global data, and I find that pages often end up with the title from a different page.

**meta-title.webc**
```html
<script webc:setup>
const metaTitle = page.title ? `${page.title} - ${site.title}` : site.title;
</script>
<title @text="metaTitle"></title>
```

On a page with no `title` frontmatter—such as my home page—I expect it to get the site title only, but it typically ends up with the page title from another page: “Some Blog Post - Site Title”.

I’m unsure if this is a bug, or if the issue is that the setup gets run once and has access to whatever data is in the cascade the first time it’s used, and then that data hangs around for subsequent executions.

**Comment by unonweb** [comment]
I'm adding the link to the issue that you've filed, because I think that you've made some good guesses:
https://github.com/11ty/eleventy-plugin-webc/issues/69

a possibly related snippet in the docs: https://www.11ty.dev/docs/languages/webc/#rendering-modes
